src/datastore/pg-write-store.ts

```diff
@@ -727,6 +731,10 @@ export class PgWriteStore extends PgStore {
         );
       }
 
+      if (!this.isEventReplay) {
+        await this.reconcileMempoolStatus(sql);
```

Comment:
@rafaelcr do these 2 call sites look safe to you?

Reply:
Yep, it's inside the write tx so I think we're good.